src/components/Badge/Badge.tsx

```diff
@@ -13,14 +13,41 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import React, {
+  forwardRef,
+  MouseEvent,
+  KeyboardEvent,
+  Ref,
+  HTMLProps
+} from 'react';
 import { css } from '@emotion/core';
 import isPropValid from '@emotion/is-prop-valid';
 
+import styled, { StyleProps } from '../../styles/styled';
 import { subHeadingKilo, focusOutline } from '../../styles/style-helpers';
 
+type OnClick = (
+  event: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>
+) => void;
+type RefType = Ref<HTMLDivElement>;
+
+export interface BadgeProps extends HTMLProps<HTMLDivElement> {
+  /**
+   * Callback for the click event.
+   */
+  onClick?: OnClick;
+  /**
+   * Ensures text is centered and the badge looks like a circle.
+   */
+  circle?: boolean;
+  color?: 'neutral' | 'primary' | 'success' | 'warning' | 'danger';
```

Comment:
[Previously](https://github.com/sumup-oss/circuit-ui/blob/4675ef115f40411afe48fe7984eb3fb00012975e/src/components/Heading/Heading.tsx#L22) we've declared a type for this, i.e.:

```typescript
type Color = 'neutral' | 'primary' | 'success' | 'warning' | 'danger';

interface BadgeProps {
  color?: Color;
}
```

But I think it's fine (and arguably more readable?) to keep it in the interface directly.

The prop should have a description, though. I know it wasn't there before either. It can be something simple, like `The semantic color of the badge`.